### DISCUSSION TITLE
[DOCS] Clearer documentation for masterless salt minions with use of beacons and reactors

### DISCUSSION THREAD

**Discussion by Anthonyhawkins** [saltstack/salt#67315] [Feature Requests]
**Description**
I'm working on trying to get a masterless salt minion running which can monitor files via a beacon, and then react to those events via a reactor.  All the documentation seems to be geared towards running the beacons and reactors with a salt master.  

**Suggested Fix**
I expected to see a clear example of using beacons and reactors with masterless salt, or at least a warning if what I'm trying to do isn't possible.   There doesn't seem to be any docs or mentioning of what this `__beacons_return` tag when viewed from salt-minion event bus.

**Type of documentation**
Masterless guides 

**Location or format of documentation**
[Beacon Example](https://docs.saltproject.io/en/latest/topics/beacons/index.html#beacon-example)

*Note, the doc mentioned to use inotify, I swapped that out for watchdog. 

**Additional context**
For example the guide suggest running `salt-run state.event pretty=true` to see the beacon event
Given I'm don't have a master I run `salt-call --local state.event pretty=True`

The docs show this
```
{
 "_stamp": "2015-09-09T15:59:37.972753",
 "data": {
     "change": "IN_IGNORED",
     "id": "larry",
     "path": "/etc/important_file"
 },
 "tag": "salt/beacon/larry/inotify//etc/important_file"
}
```

I see this which I can't figure out how to have the reactor react to. 
```
__beacons_return	{
    "_stamp": "2022-09-16T02:38:17.552441",
    "beacons": [
        {
            "beacon_name": "watchdog",
            "data": {
                "change": "modified",
                "id": "my-host",
                "path": "/etc/important_file"
            },
            "tag": "salt/beacon/my-host/important_file/watchdog"
        },
        {
            "beacon_name": "watchdog",
            "data": {
                "change": "modified",
                "id": "my-host",
                "path": "/etc/important_file"
            },
            "tag": "salt/beacon/my-host/important_file/watchdog"
        },
        {
            "beacon_name": "watchdog",
            "data": {
                "change": "modified",
                "id": "my-host",
                "path": "/etc"
            },
            "tag": "salt/beacon/my-host/important_file/watchdog"
        }
    ]
}
```

I appreciate any help, thank you.

**Comment by Anthonyhawkins** [comment]
I'm still digging through this, it seems it may need a code change.  in `minion.py` here, 
https://github.com/saltstack/salt/blob/1a546537b1f0810e0b576986ad54d52fde2d539a/salt/minion.py#L2986-L2997

I modified this to check if the `master_type` is `disable` for a master-less minion scenario, and fire off each beacon as a separate event vs the `__beacons_return` event.

```python
            def handle_beacons():
                # Process Beacons
                beacons = None
                try:
                    beacons = self.process_beacons(self.functions)
                except Exception:  # pylint: disable=broad-except
                    log.critical("The beacon errored: ", exc_info=True)
                if beacons:
                    with salt.utils.event.get_event(
                        "minion", opts=self.opts, listen=False
                    ) as event:
                        if self.opts["master_type"] == "disable":
                            for beacon in beacons:
                                event.fire_event(beacon["data"], beacon["tag"])
                        else:
                            event.fire_event({"beacons": beacons}, "__beacons_return") 
```

This will now allow the beacons to present themselves on the minion's event bus with their top-level tags so they can be picked up on by the minion's reactor etc. I'm still testing it though so I'm not sure if this will create any other oddities further down the road.

**Comment by barbaricyawps** [comment]
Thanks for opening this issue! I'm removing the Docs label and adding the Feature label since this request will require a code change. I'll leave the Needs Triage label on for now so that it can be evaluated by the rest of the team.
